### ISSUE TITLE
ValueError when using parse_yokogawa_metadata on fake metadata files

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal#130]
Within a test, we have a fake dataset formed by two images :
```
myplate_B03_T0002F001L01A01Z01C01.png
myplate_B03_T0002F002L01A01Z01C01.png
```
generated with
```python
import numpy as np
from PIL import Image

im1 = np.arange(2560 * 2160, dtype=np.uint16).reshape(2160, 2560)
im2 = np.arange(2560 * 2160, dtype=np.uint16).reshape(2560, 2160).T
Image.fromarray(im1).save(
    "myplate_B03_T0002F001L01A01Z01C01.png", optimize=True
)
Image.fromarray(im2).save(
    "myplate_B03_T0002F002L01A01Z01C01.png", optimize=True
)
```

And then there are the corresponding mlf/mrf files:
```xml
<?xml version="1.0" encoding="utf-8"?>
<bts:MeasurementData bts:Version="1.0" xmlns:bts="http://www.yokogawa.co.jp/BTS/BTSSchema/1.0">
<bts:MeasurementRecord bts:Type="IMG" bts:Time="2020-08-12T17:36:36.234+02:00" bts:Column="3" bts:Row="2" bts:TimePoint="2" bts:FieldIndex="1" bts:ZIndex="1" bts:TimelineIndex="1" bts:ActionIndex="1" bts:Action="3D" bts:X="-1448.3" bts:Y="1517.7" bts:Z="-2.0" bts:Ch="1">myplate_B03_T0002F001L01A01Z01C01.tif</bts:MeasurementRecord>
<bts:MeasurementRecord bts:Type="IMG" bts:Time="2020-08-12T17:36:46.322+02:00" bts:Column="3" bts:Row="2" bts:TimePoint="2" bts:FieldIndex="2" bts:ZIndex="1" bts:TimelineIndex="1" bts:ActionIndex="1" bts:Action="3D" bts:X="-1032.3" bts:Y="1517.7" bts:Z="-2.0" bts:Ch="1">myplate_B03_T0002F002L01A01Z01C01.tif</bts:MeasurementRecord>
</bts:MeasurementData>
```
```xml
<?xml version="1.0" encoding="utf-8"?>
<bts:MeasurementDetail bts:Version="1.0" bts:OperatorName="Joel" bts:Title="20200812-CardiomyocyteDifferentiation14-Cycle1" bts:Application="" bts:BeginTime="2020-08-12T17:27:17.603876+02:00" bts:EndTime="2020-08-13T03:20:17.1789113+02:00" bts:MeasurementSettingFileName="20200812-Joel-CardiomyocyteDifferentiation14-Cycle1.mes" bts:ColumnCount="12" bts:RowCount="8" bts:TimePointCount="5" bts:FieldCount="72" bts:ZCount="42" bts:TargetSystem="CV7000 FD" bts:ReleaseNumber="R1.17.05" xmlns:bts="http://www.yokogawa.co.jp/BTS/BTSSchema/1.0">
  <bts:MeasurementSamplePlate bts:Name="20200812-CardiomyocyteDifferentiation14-Cycle1" bts:WellPlateFileName="20200812-CardiomyocyteDifferentiation14-Cycle1.wpi" bts:WellPlateProductFileName="1009602002_Greiner_#655090.wpp" />
  <bts:MeasurementChannel bts:Ch="1" bts:HorizontalPixelDimension="0.1625" bts:VerticalPixelDimension="0.1625" bts:CameraNumber="3" bts:InputBitDepth="16" bts:InputLevel="1000" bts:HorizontalPixels="2560" bts:VerticalPixels="2160" bts:FilterWheelPosition="6" bts:FilterPosition="1" bts:ShadingCorrectionSource="SC_BP445-45_40x_M10_CH01.tif" />
</bts:MeasurementDetail>
```

@jluethi, do you see anything weird in there?

When I run your parsing function as in
```python
from fractal.tasks.metadata_parsing import parse_yokogawa_metadata
parse_yokogawa_metadata("MeasurementDetail.mrf", "MeasurementData.mlf")
```
I get this ValueError:
```python traceback
Traceback (most recent call last):
  File "test_read.py", line 4, in <module>
    parse_yokogawa_metadata("MeasurementDetail.mrf", "MeasurementData.mlf")
  File "/net/nfs4/pelkmanslab-fileserver-common/data/homes/fractal/mwe_fractal/fractal/tasks/metadata_parsing.py", line 60, in parse_yokogawa_metadata
    site_metadata = pd.concat([site_metadata, get_z_steps(mlf_frame)], axis=1)
  File "/net/nfs4/pelkmanslab-fileserver-common/data/homes/fractal/mwe_fractal/fractal/tasks/metadata_parsing.py", line 284, in get_z_steps
    z_data = grouped_sites_z.apply(calculate_steps).groupby(
  File "/data/homes/fractal/.conda/envs/fractal/lib/python3.8/site-packages/pandas/core/frame.py", line 7718, in groupby
    return DataFrameGroupBy(
  File "/data/homes/fractal/.conda/envs/fractal/lib/python3.8/site-packages/pandas/core/groupby/groupby.py", line 882, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/data/homes/fractal/.conda/envs/fractal/lib/python3.8/site-packages/pandas/core/groupby/grouper.py", line 872, in get_grouper
    obj._check_label_or_level_ambiguity(gpr, axis=axis)
  File "/data/homes/fractal/.conda/envs/fractal/lib/python3.8/site-packages/pandas/core/generic.py", line 1794, in _check_label_or_level_ambiguity
    raise ValueError(msg)
ValueError: 'well_id' is both an index level and a column label, which is ambiguous.
```

**Comment by tcompa** [comment]
Quick note: in the mrf I also have to rename the plate to `myplate` (which in the previous comment was left unchanged), but that doesn't change the error.

**Comment by jluethi** [comment]
I'll look into it. Sounds like an issue with the aggregation of things within the metadata parsing function. I should really write some tests to ensure the different parts remain stable.

Weird that this bug doesn't arise when parsing normal experimental data, but good that we found an edge case I guess.
I will debug this with the notebooks I created while building the parsing, that will be easier than debugging it all at once.

**Comment by tcompa** [comment]
Tests are definitely useful, but be aware that I just copy-pasted-modified some existing mlf/mrf files and I'm not 100% sure of what I was doing (AKA please check that the two files in the comment above look correct).

**Comment by jluethi** [comment]
@tcompa I can reproduce this bug. It's a bug that appears when there is a single Z plane per FOV. So the metadata itself is fine and should be supported, but the current parsing doesn't handle this well.
I will look into a fix and work on testing the parsing as well. If you need to work around this and the test can also consist of multiple z planes, this metadata for 2 planes per FOV is parsed correctly.

```
<?xml version="1.0" encoding="utf-8"?>
<bts:MeasurementData bts:Version="1.0" xmlns:bts="http://www.yokogawa.co.jp/BTS/BTSSchema/1.0">
<bts:MeasurementRecord bts:Type="IMG" bts:Time="2020-08-12T17:36:36.234+02:00" bts:Column="3" bts:Row="2" bts:TimePoint="2" bts:FieldIndex="1" bts:ZIndex="1" bts:TimelineIndex="1" bts:ActionIndex="1" bts:Action="3D" bts:X="-1448.3" bts:Y="1517.7" bts:Z="-2.0" bts:Ch="1">myplate_B03_T0002F001L01A01Z01C01.tif</bts:MeasurementRecord>
<bts:MeasurementRecord bts:Type="IMG" bts:Time="2020-08-12T17:36:36.234+02:00" bts:Column="3" bts:Row="2" bts:TimePoint="2" bts:FieldIndex="1" bts:ZIndex="2" bts:TimelineIndex="1" bts:ActionIndex="1" bts:Action="3D" bts:X="-1448.3" bts:Y="1517.7" bts:Z="-1.0" bts:Ch="1">myplate_B03_T0002F001L01A01Z02C01.tif</bts:MeasurementRecord>
<bts:MeasurementRecord bts:Type="IMG" bts:Time="2020-08-12T17:36:46.322+02:00" bts:Column="3" bts:Row="2" bts:TimePoint="2" bts:FieldIndex="2" bts:ZIndex="1" bts:TimelineIndex="1" bts:ActionIndex="1" bts:Action="3D" bts:X="-1032.3" bts:Y="1517.7" bts:Z="-2.0" bts:Ch="1">myplate_B03_T0002F002L01A01Z01C01.tif</bts:MeasurementRecord>
<bts:MeasurementRecord bts:Type="IMG" bts:Time="2020-08-12T17:36:46.322+02:00" bts:Column="3" bts:Row="2" bts:TimePoint="2" bts:FieldIndex="2" bts:ZIndex="2" bts:TimelineIndex="1" bts:ActionIndex="1" bts:Action="3D" bts:X="-1032.3" bts:Y="1517.7" bts:Z="-1.0" bts:Ch="1">myplate_B03_T0002F002L01A01Z02C01.tif</bts:MeasurementRecord>
</bts:MeasurementData>
```

**Comment by tcompa** [comment]
Ok, thanks.

As a quick fix, I am running tests on a 2-FOVs and 2-Z-planes dataset (https://github.com/fractal-analytics-platform/fractal/commit/19c48b5a0eb50a51353f0604a849d2458b44cd27 and https://github.com/fractal-analytics-platform/fractal/commit/e57bf1b4e887f8049fc159677b4b62ad026aa077).

**Comment by jluethi** [comment]
With 50ffc46ea8b327182c779a32876104dde84f1f18, I now added functionality to parse metadata with only a single Z plane and made the parsing a bit more robust.
f3e1119a8f01b0632be8c23361b11a78035275b7 adds some tests to ensure the correct metadata is parsed. When merging, I saw that you also added a first set of tests in `test_unit_parse_yokogawa_metadata.py`. @tcompa I think we can merge them. I also test that the number of lines are parsed correctly, plus that the columns contain the correct data.
I want to improve the tests a bit further while I'm at it and will close the issue once the test coverage is a bit broader.
